src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryCustomObject.cs

```diff
@@ -124,12 +131,55 @@ public static FormulaValue Index_CO(IRContext irContext, FormulaValue[] args)
 
         public static FormulaValue Value_CO(EvalVisitor runner, SymbolContext symbolContext, IRContext irContext, CustomObjectValue[] args)
         {
-            return new NumberValue(irContext, args[0].Impl.GetDouble());
+            var impl = args[0].Impl;
+            double number;
+
+            if (impl.IsString)
+            {
+                if (!double.TryParse(impl.GetString(), out number))
+                {
+                    return CommonErrors.InvalidNumberFormatError(irContext);
+                }
+            }
+            else if (impl.IsNull)
+            {
+                return new BlankValue(irContext);
+            }
+            else if (impl.IsBoolean)
+            {
+                number = impl.GetBoolean() ? 1 : 0;
+            }
+            else
+            {
+                number = impl.GetDouble();
+            }
+
+            return new NumberValue(irContext, number);
         }
 
         public static FormulaValue Text_CO(EvalVisitor runner, SymbolContext symbolContext, IRContext irContext, CustomObjectValue[] args)
         {
-            return new StringValue(irContext, args[0].Impl.GetString());
+            var impl = args[0].Impl;
+            string str;
+
+            if (impl.IsString)
+            {
+                str = impl.GetString();
+            }
+            else if (impl.IsNull)
+            {
+                str = string.Empty;
+            }
+            else if (impl.IsBoolean)
+            {
+                str = impl.GetBoolean() ? "true" : "false";
```

Comment:
> str = impl.GetBoolean() ? "true" : "false"; [](http://example.com/codeflow?start=15&length=44)

Can (should?) this chain to the existing Text function?
Ensure we get details right (like whether "true" is upper or lower; locales, etc. 
That provides a path for handling format string.

Reply:
Are you saying we need to construct a new BooleanValue to pass into the existing coercion?